src/main/java/com/nvidia/spark/rapids/jni/kudo/SerializedTableHeader.java

```diff
@@ -0,0 +1,155 @@
+package com.nvidia.spark.rapids.jni.kudo;
+
+import java.io.DataInputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * Holds the metadata about a serialized table. If this is being read from a stream
+ * isInitialized will return true if the metadata was read correctly from the stream.
+ * It will return false if an EOF was encountered at the beginning indicating that
+ * there was no data to be read.
+ */
+public final class SerializedTableHeader {
+    /**
+     * Magic number "KUDO" in ASCII.
+     */
+    private static final int SER_FORMAT_MAGIC_NUMBER = 0x4B55444F;
+    private static final short VERSION_NUMBER = 0x0001;
+
+    // Useful for reducing calculations in writing.
+    private long offset;
+    private long numRows;
```

Comment:
It's impossible to have more than a 31-bit row count, this should just be an int.